nix/eval-machine-info.nix

```
@@ -11,104 +11,136 @@
 let
   call = x: if builtins.isFunction x then x args else x;
 
-  # Copied from nixpkgs to avoid <nixpkgs> import
-  optional = cond: elem: if cond then [elem] else [];
-
   zipAttrs = set: builtins.listToAttrs (
     map (name: { inherit name; value = builtins.catAttrs name set; }) (builtins.concatMap builtins.attrNames set));
 
-  flakeExpr = (builtins.getFlake flakeUri).outputs.nixopsConfigurations.default;
-
-  networks =
-    let
-      getNetworkFromExpr = networkExpr:
-        (call (import networkExpr)) // { _file = networkExpr; };
+  flake = builtins.getFlake flakeUri;
+  flakeExpr = (builtins.tryEval flake.outputs.nixopsConfigurations.default).value or { };
 
-      exprToKey = key: { key = toString key; };
-
-      networkExprClosure = builtins.genericClosure {
-        startSet = map exprToKey networkExprs;
-        operator = { key }: map exprToKey ((getNetworkFromExpr key).require or []);
-      };
-    in
-      map ({ key }: getNetworkFromExpr key) networkExprClosure
-      ++ optional (flakeUri != null)
-        ((call flakeExpr) // { _file = "<${flakeUri}>"; });
+  nixpkgs1 = <nixpkgs>; # this will be replaced on install by nixops' nixpkgs input
+  lib = import "${nixpkgs1}/lib";
```

We should use this exclusively for getting the `nixpkgs` option value, and then use its lib for the rest of the evaluation.

```suggestion
  libBootNixpkgs = <nixpkgs>; # this will be replaced on install by nixops' nixpkgs input
  libBootLib = import "${nixpkgs1}/lib";
```